### PR TITLE
Fix '%f' specifier for date_format and date_parse

### PR DESCRIPTION
**Pull request by haozhun**
#2590

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java [hunk 1]
```diff
@@ -591,14 +591,14 @@ public void testDateFormat()
         assertFunction("date_format(" + dateTimeLiteral + ", '%4')", "4");
         assertFunction("date_format(" + dateTimeLiteral + ", '%x %v')", "2001 02");
 
-        String wierdDateTimeLiteral = "TIMESTAMP '2001-01-09 13:04:05.321 +07:09'";
+        String wierdDateTimeLiteral = "TIMESTAMP '2001-01-09 13:04:05.32 +07:09'";
```
Comment (dain) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java]: why drop the extra digit?

**Comment by electrum** [comment]
Title the commit message "Fix '%f' specifier for date_format and date_parse"

**Comment by electrum** [comment]
Add release notes

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java (hunk 1 above)
Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java]: `321` is easier to get right than `32`. You can't get `32` right unless you are counting from left to right. But you can luckily get `321` right if you are parsing it as millisecond.

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java [hunk 2]
```diff
@@ -642,6 +642,10 @@ public void testDateParse()
         assertFunction("date_parse('2013 14', '%Y %y')", toTimestamp(new DateTime(2014, 1, 1, 0, 0, 0, 0, DATE_TIME_ZONE)));
 
         assertFunction("date_parse('1998 53', '%x %v')", toTimestamp(new DateTime(1998, 12, 28, 0, 0, 0, 0, DATE_TIME_ZONE)));
+
+        assertFunction("date_parse('1.1', '%s.%f')", toTimestamp(new DateTime(1970, 1, 1, 0, 0, 1, 100, DATE_TIME_ZONE)));
+        assertFunction("date_parse('1.01', '%s.%f')", toTimestamp(new DateTime(1970, 1, 1, 0, 0, 1, 10, DATE_TIME_ZONE)));
+        assertFunction("date_parse('1.2006', '%s.%f')", toTimestamp(new DateTime(1970, 1, 1, 0, 0, 1, 200, DATE_TIME_ZONE)));
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java]: Is truncation the correct behavior?

**Comment by haozhun** [comment]
Fixed issues mentioned in comments from @dain and @electrum .

**Comment by electrum** [comment]
Looks good

File: presto-docs/src/main/sphinx/release/release-0.100.rst [hunk 3]
```diff
@@ -4,5 +4,6 @@ Release 0.100
 
 General Changes
 ---------------
+* Fix a bug for '%f' specifier in :func:`date_format` and :func:`date_parse`
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.100.rst]: Remove "a bug for"

Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.100.rst]: "Fix '%f' specifier for ..."

Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.100.rst]: Add a period at the end.

**Comment by dain** [comment]
Looks good

**Pull request merged**